### DISCUSSION TITLE
Make CRUD view URLs more user-friendly

### DISCUSSION THREAD

**Discussion by ktmud** [apache/superset#18416] [Ideas]
## Screenshot

<img src="https://user-images.githubusercontent.com/335541/98577919-7ab0fb80-2271-11eb-8020-b1ca291a7a74.png" width="500" />

## Description

As a professional software, the URLs should be intuitive and predictable. Currently most CRUD view pages have automatically-generated URL pathnames which looks kind of hacky. We should change them to something more user-friendly, let them follow the Superset nomenclature instead the internal Python model/view names.

```
Datasets: /tablemodelview/ -> /datasets
Databases: /databaseview/ -> /databases
```

## Design input
[describe any input/collaboration you'd like from designers, and
tag accordingly. For design review, add the
label `design:review`. If this includes a design proposal,
include the label `design:suggest`]

[a comment by an automated account is not shown]

**Comment by nytai** [comment]
This has been bothering me too. There's some legacy here, as you've pointed out regarding 
> internal Python model/view names

Listing out what would be required to make this happen:
- Set up redirects for the old routes (in case users have bookmarked these urls)
- set up backend routing for these new routes 
- set up react-router routes for the new routes

* There's likely something else I'm missing.

We should wait until the react crud views are complete and we can fully deprecate the old FAB views since I don't think there's a way to change the routes for the old FAB views.
